Sources/Environment.swift

```diff
@@ -15,6 +15,11 @@ public extension Extension {
 
   // MARK: - Private
 
+  private func registerBooleanFilterWithArguments(_ name: String, filter: @escaping Filters.BooleanWithArguments) {
```

Comment:
Just nitpicking, but I wonder if we shouldn't just call this method `registerFilter` like all the others — and let the type overloading do the magic of using the proper version of `registerFilter` depending on the kind of filter passed…

Reply:
@Antondomashnev up to you

Reply:
Hah, I've tried it but I forgot that call original `registerFilter` inside the `registerBooleanFilterWithArguments ` and it leads to the recursion 😞 @krzysztofzablocki @AliSoftware do you have any idea how to workaround it?

Reply:
Using `as` to specify the overload you want to call should do it:

```swift
typealias AnyFilter = (Any?, [Any?]) throws -> Any?
registerFilter(name, filter: filter as AnyFilter)
registerFilter("!\(name)", filter: { value, arguments in try !filter(value, arguments) } as AnyFilter)
```